tests/ci/validate-specs.test.sh

```
@@ -0,0 +1,238 @@
+#!/bin/bash
+# Integration tests for spec validation script
+
+set -euo pipefail
+
+readonly SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)